spec/importc.dd

```diff
@@ -4,7 +4,166 @@ $(SPEC_S ImportC,
+    which will hereafter be referred to as $(B C11).
+    References to the C11 Standard will be C11 followed by the paragraph number.
+    Prior versions, such as C99, C89, and K+R C, are not supported.
+    )
+
+    $(P Adjustment to the ImportC dialect is made to match the
+    behavior of the C compiler that the D compiler is matched to. For
+    example, on Win32 D is matched to the Digital Mars C compiler,
+    and can be matched to the Visual C compiler using the $(TT -mscoff32)
+    switch. Win64 D is matched to the Visual C compiler.
+    On Posix targets, the matching C compiler is Gnu C or Clang C.
+    )
+
+    $(P Further adjustment is made to take advantage of some of the D
+    implementation's capabilities.)
+
+    $(P This is all covered in the rest of this document.)
+
+
+$(H2 $(LNAME2 command-line, Invoking ImportC))
+
+    $(P The ImportC compiler can be invoked in two ways:)
+
+    $(UL
+    $(LI directly via the command line)
+    $(LI indirectly via importing a C file)
+    )
+
+    $(H3 ImportC Files on the Command Line)
+
+    $(H3 Importing C Files)
+
+
+$(H2 $(LNAME2 preprocessor, Preprocessor))
+
+    $(P ImportC does not have a preprocessor. It is designed to compile C
+    files after they have been first run through the C preprocessor.)
+
+    $(H3 Digital Mars C Preprocessor)
+
+    $(H3 Gnu C Preprocessor)
+
+    $(H3 Clang C Preprocessor)
+
+    $(H3 Warp C Preprocessor)
+
+
+$(H2 $(LNAME2 preprocessor-directives, Preprocessor Directives))
+
+    $(P Nevertheless, ImportC supports these preprocessor directives:)
+
+    $(H3 Line control)
+
+    $(P C11 6.10.4)
+
+    $(H3 Linemarker)
+
+    $(P $(LINK2 https://gcc.gnu.org/onlinedocs/gcc-11.1.0/cpp/Preprocessor-Output.html, linemarker)
+    directives are normally embedded in the output of C preprocessors.)
+
+$(H2 $(LINK2 limitations, Limitations))
+
+    $(H3 Exception Handling)
+
+    $(P ImportC is assumed to never throw exceptions. `setjmp` and `longjmp` are not supported.)
+
+    $(H3 Transitive Const)
+
+    $(P C11 specifies that `const` only applies locally. `const` in ImportC applies transitively,
+    meaning that although $(CCODE int *const p;) means in C11 that p is a const pointer to int,
+    in ImportC it means p is a const pointer to a const int.)
+
+$(H2 $(LINK2 extensions, Extensions))
+
+
+$(H2 $(LINK2 gnu-clang-extensions, Gnu and Clang Extensions))
+
+
+$(H2 $(LINK2 visualc-extensions, Visual C Extensions))
+
+
+$(H2 $(LINK2 digital-mars-extensions, Digital Mars C Extensions))
+
+
+$(H2 $(LINK2 d-side, ImportC from D's Point of View))
+
+    $(P There is no one-to-one mapping of C constructs to D constructs, although
+    it is very close. What follows is a description of how the D side views
+    the C declarations that are imported.)
+
+    $(H3 Module Name)
+
+    $(P The module name assigned to the ImportC file is the filename stripped
+    of its path and extension. This is just like the default module name assigned
```

Comment:
Do we need to reassure that same name, different directory won't conflict? I.e importing both math.h and bits/math.h.

Reply:
It works exactly the same way as the default module name. It's the same code. I didn't change a line of code for ImportC.